src/public/modules/forms/admin/views/collaborator.client.modal.html

```diff
@@ -0,0 +1,341 @@
+                  permissions below.</span
+                >
+              </div>
+            </div>
+            <!-- Viewer mode only dropdown menu-->
+            <div
+              id="collab-role"
+              ng-if="userCanEdit && myform.responseMode === 'encrypt'"
+              class="col-xs-12 col-sm-3 role-column"
+            >
+              <div
+                id="new-collaborator-dropdown"
+                class="btn-group btn-block"
+                uib-dropdown
+              >
+                <button
+                  id="change-role-button"
+                  type="button"
+                  uib-dropdown-toggle
+                  class="input-custom input-medium role-dropdown-btn"
+                >
+                  <span class="pull-left">{{ collab.form.role }}</span>
+                  <span class="caret pull-right"></span>
+                </button>
+                <ul
+                  id="role-dropdown-menu"
+                  class="dropdown-menu dropdown-menu-right"
+                  role="menu"
+                  aria-labelledby="change-role-button"
+                  uib-dropdown-menu
+                >
+                  <li role="menuitem">
+                    <a
+                      href
+                      ng-if="user.email === myform.admin.email"
+                      ng-click="selectRole(ROLES.ADMIN)"
+                    >
+                      <div class="input-medium">{{ ROLES.ADMIN }}</div>
+                      <div class="role-explanation">
+                        Can fully configure and edit
+                      </div>
+                    </a>
+                  </li>
+                  <li class="spacer"></li>
+                  <li role="menuitem">
+                    <a href ng-click="selectRole(ROLES.EDITOR)">
+                      <div class="input-medium">{{ ROLES.EDITOR }}</div>
+                      <div class="role-explanation">
+                        Can edit but can't delete forms
+                      </div>
+                    </a>
+                  </li>
+                  <li class="spacer"></li>
+                  <li role="menuitem">
+                    <a href ng-click="selectRole(ROLES.VIEWER)">
+                      <div class="input-medium">{{ ROLES.VIEWER }}</div>
+                      <div class="role-explanation">
+                        Can only view responses
+                      </div>
+                    </a>
+                  </li>
+                </ul>
+              </div>
+            </div>
+          </div>
+          <!-- Reminder to share secret key -->
+          <div
+            class="alert-custom alert-info secret-key-warning"
+            ng-if="myform.responseMode === 'encrypt'"
+          >
+            <i class="bx bx-info-circle bx-md icon-spacing"></i>
+            <span class="alert-msg">
+              Collaborators need the Secret Key to access form responses. Share
```

Comment:
some of this code is duplicated in `admin-form.client.view.html`. There might be some controller code duplicated too. Could you clean up those?

Reply:
done